dbt/include/redshift/macros/adapters.sql

```diff
@@ -1,6 +1,10 @@
 
 {% macro dist(dist) %}
   {%- if dist is not none -%}
+      {%- if dist is iterable or dist is mapping -%}
+        {% exceptions.raise_compiler_error("Expected a single valued property for dist and got: " ~ dist ~ " instead") %}
```

Comment:
I think this needs a 'do' in front of it, e.g.:
```
{% do exceptions.raise_compiler_error("Expected a single valued property for dist and got: " ~ dist ~ " instead") %}
```
Alternatively, I think `{{` instead of `{%` also works:
```
{{ exceptions.raise_compiler_error("Expected a single valued property for dist and got: " ~ dist ~ " instead") }}
```

Reply:
Yeah that's my mistake on the compiler error- it needs a do statement. Let me update this later today. There's an issue with the conditional check. Strings are technically iterable so this isn't doing what I want it to do in any case